airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt

```diff
@@ -490,42 +358,27 @@ class SnowflakeSqlGenerator(private val retentionPeriodDays: Int) : SqlGenerator
         primaryKey: List<ColumnId>,
         cursor: Optional<ColumnId>
     ): String {
-        val pkList =
-            primaryKey
-                .stream()
-                .map { columnId: ColumnId -> columnId.name(QUOTE) }
-                .collect(Collectors.joining(","))
+        val pkList = primaryKey.joinToString { it.name(QUOTE) }
         val cursorOrderClause =
             cursor
-                .map { cursorId: ColumnId -> cursorId.name(QUOTE) + " DESC NULLS LAST," }
+                .map { cursorId: ColumnId -> "${cursorId.name(QUOTE)} DESC NULLS LAST," }
                 .orElse("")
-
-        return StringSubstitutor(
-                java.util.Map.of(
-                    "final_table_id",
-                    id.finalTableId(QUOTE, finalSuffix.uppercase(Locale.getDefault())),
-                    "pk_list",
-                    pkList,
-                    "cursor_order_clause",
-                    cursorOrderClause,
-                    "airbyte_extracted_at_utc",
-                    airbyteExtractedAtUtcForced("\"_AIRBYTE_EXTRACTED_AT\"")
-                )
-            )
-            .replace(
-                """
-            DELETE FROM ${'$'}{final_table_id}
-            WHERE "_AIRBYTE_RAW_ID" IN (
-              SELECT "_AIRBYTE_RAW_ID" FROM (
-                SELECT "_AIRBYTE_RAW_ID", row_number() OVER (
-                  PARTITION BY ${'$'}{pk_list} ORDER BY ${'$'}{cursor_order_clause} ${'$'}{airbyte_extracted_at_utc} DESC
-                ) as row_number FROM ${'$'}{final_table_id}
-              )
-              WHERE row_number != 1
-            );
-            
-            """.trimIndent()
-            )
+        return """
+        |DELETE FROM 
+        |  ${id.finalTableId(QUOTE, finalSuffix.uppercase(Locale.getDefault()))}
```

Comment:
when the same expression is several times in the same string (or across several strings), I'd rather see it in its own variable. I think it also makes the string template easier to read

Reply:
we could also add a function that does the `id.finalTableId.etc1

Reply:
Yup. there are many such duplicated snippets. Tried avoiding them but these little ones probably won't hurt much.

Reply:
fixed this particular one. I'll do more cleanup when introducing `MERGE`